geopandas/sindex.py

```diff
@@ -625,13 +625,19 @@ def __len__(self):
             return self.size
 
 
-if compat.HAS_PYGEOS:
+if compat.USE_SHAPELY_20 or compat.HAS_PYGEOS:
```

Comment:
```suggestion
if compat.USE_SHAPELY_20 or compat.USE_PYGEOS:
```

Reply:
This is run on import of geopandas, at which point it might not yet be known if pygeos will be used or not, so we can at least define the class always if pygeos is availble (so HAS_PYGEOS is ok here).

Of course, that explanation is also valid for USE_SHAPELY_20 .., so I suppose I can also just remove that here and always define this class if pygeos is available (it's not that it is overwriting something else, it's just defining an additional class that is available)

Reply:
Hmm, this actually points out a problem, because after this line, we _do_ decide to import pygeos vs shapely based on `USE_SHAPELY_20`/`USE_PYGEOS`. 
So that means you can't easily switch between shapely 2.0 and pygeos after import (and so you have to use the environment variable, and don't switch in the middle of a (test) session when using the sindex)

Reply:
Other than for purposes of tests, it seems OK that you cannot change between shapely 2.0 or pygeos after initialization.  We just probably need to document that somewhere.

Reply:
Pushed some documentation updates for the section about using PyGEOS with geopandas.